### PR TITLE
[REVIEW] Change default value of `ordered` to `False` in `CategoricalDtype`

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
Fixes: #11487 

This PR switches default value of `ordered` parameter in `CategoricalDtype` to `False`. This fixes some issues around concat and building categorical columns.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by bdice — APPROVED** [review]
Seems like a straightforward change. Thanks!

File: python/cudf/cudf/core/column/column.py [hunk 1]
```diff
@@ -1475,7 +1475,7 @@ def build_categorical_column(
     size: int = None,
     offset: int = 0,
     null_count: int = None,
-    ordered: bool = None,
+    ordered: bool = False,
```
Comment (bdice) [inline python/cudf/cudf/core/column/column.py:1478]: Oh, please update the docstring below to indicate the parameter is optional and the default is False.

File: python/cudf/cudf/core/dtypes.py [hunk 2]
```diff
@@ -125,9 +125,9 @@ class CategoricalDtype(_BaseDtype):
     stored on the GPU.
     """
 
-    ordered: Optional[bool]
+    ordered: bool
 
-    def __init__(self, categories=None, ordered: bool = None) -> None:
+    def __init__(self, categories=None, ordered: bool = False) -> None:
```
Comment (bdice) [inline python/cudf/cudf/core/dtypes.py:130]: If this API is public, we should improve its docstring to look more like https://pandas.pydata.org/docs/reference/api/pandas.CategoricalDtype.html.

File: python/cudf/cudf/core/column/column.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/column/column.py:1478]: Done

File: python/cudf/cudf/core/dtypes.py (hunk 2 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/dtypes.py:153]: Updated the docstring, but a quick lookup shows this isn't published in docs yet. Added an issue to follow-up: https://github.com/rapidsai/cudf/issues/11605

Comment (bdice) [reply python/cudf/cudf/core/dtypes.py:153]: Thanks!

File: python/cudf/cudf/core/dtypes.py [hunk 3]
```diff
@@ -121,13 +121,36 @@ class _BaseDtype(ExtensionDtype, Serializable):
...
+        Whether or not this categorical is treated as a ordered categorical.
+        None can be used to maintain the ordered value of existing categoricals
+        when used in operations that combine categoricals, e.g. astype, and
+        will resolve to False if there is no existing ordered to maintain.
+
+    Examples
+    --------
+    >>> import cudf
+    >>> dtype = cudf.CategoricalDtype(categories=['b', 'a'], ordered=True)
+    >>> cudf.Series(['a', 'b', 'a', 'c'], dtype=t)
```
Comment (galipremsagar) [inline python/cudf/cudf/core/dtypes.py:142]:
```suggestion
    >>> cudf.Series(['a', 'b', 'a', 'c'], dtype=dtype)
```

**Review by shwina — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**